lib/api.js

```diff
@@ -429,15 +420,23 @@ api.declare({
   // instance is not managed, and should not be authorized.  If there's more
   // than one, the database is not providing the relational guarantees it
   // should
-  if (!req.satisfies({region, instanceId})) { 
+  try {
+    await req.authorize({region, instanceId, hasWorkerType: false});
```

Comment:
I think we were planning to eventually only allow `authorize` once since it will eventually be a round-trip to auth service and back.  Can we do the entire check inside the switch below?

Reply:
The problem is that we'd be trading a round trip to Auth for a round trip to the Postgres database.  I'm not sure that it would be a problem, but it's not a totally free lookup either.

What's the timeline on moving to one-authorize()-per-handler?